### PR TITLE
[mtl] expose stitching and dummy encoder options

### PR DESCRIPTION
**Pull request by kvark**
These are useful to experiment with.
PR checklist:
- [ ] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [ ] tested examples with the following backends:
- [ ] `rustfmt` run on changed code

### CODE REVIEW

File: src/backend/metal/src/command.rs [hunk 1]
```diff
@@ -1577,8 +1580,10 @@ impl RawCommandQueue<Backend> for CommandQueue {
                                     cmd_buffer
                                 });
                             journal.record(&*cmd_buffer);
-                            if STITCH_DEFERRED_COMMAND_BUFFERS {
+                            if self.stich_deferred {
                                 deferred_cmd_buffer = Some(cmd_buffer);
+                            } else {
+                                cmd_buffer.commit();
```
Comment (grovesNL) [inline src/backend/metal/src/command.rs]: Previously a bug?

File: src/backend/metal/src/command.rs [hunk 2]
```diff
@@ -1491,6 +1485,15 @@ impl CommandQueue {
             } else {
                 None
             },
+            stich_deferred: true,
```
Comment (grovesNL) [inline src/backend/metal/src/command.rs]: `stitch_spelling` (missing `t`)

File: src/backend/metal/src/command.rs (hunk 1 above)
Comment (kvark) [reply src/backend/metal/src/command.rs]: yes :) it took me a bit of head scratching to figure out

**Comment by kvark** [comment]
All fixed, thanks for the review!
bors r=grovesNL

**Pull request merged**